pkg/api/v1/summary_test.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2021 The TestGrid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"testing"
+
+	apipb "github.com/GoogleCloudPlatform/testgrid/pb/api/v1"
+	configpb "github.com/GoogleCloudPlatform/testgrid/pb/config"
+	summarypb "github.com/GoogleCloudPlatform/testgrid/pb/summary"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+func TestListTabSummaries(t *testing.T) {
+	tests := []struct {
+		name        string
+		config      map[string]*configpb.Configuration
+		summaries   map[string]*summarypb.DashboardSummary
+		req         *apipb.ListTabSummariesRequest
+		want        *apipb.ListTabSummariesResponse
+		expectError bool
+	}{
+		{
+			name: "Returns an error when there's no dashboard in config",
+			config: map[string]*configpb.Configuration{
+				"gs://default/config": {},
+			},
+			req: &apipb.ListTabSummariesRequest{
+				Dashboard: "missing",
+			},
+			expectError: true,
+		},
+		{
+			name: "Returns empty tab summaries list for non-existing summary",
```

Comment:
This test returns an empty tab summary for an empty summary file. How would your code react if the file actually wasn't there? (ex: a tab is added to config and requested before the summarizer has summarized it for the first time)

Reply:
if summary doesn't exist yet, the error is returned. This follows the same path as ListHeader call. I added the test case to explicitly check that